scanner/scanner.go

```diff
@@ -220,3 +226,58 @@ func (s *Scanner) GitlabciConfigs() ([]models.GitlabciConfig, error) {
 
 	return configs, nil
 }
+
+var azurePipelineFileRegex = regexp.MustCompile(`\.?azure-pipelines(-.+)?\.ya?ml$`)
+
+func (s *Scanner) AzurePipelines() ([]models.AzurePipeline, error) {
+	pipelines := []models.AzurePipeline{}
+	err := filepath.Walk(s.Path,
```

Comment:
^ Note to self and @SUSTAPLE117 , we should consider optimizing this because we now have a 4th `filepath.Walk()`. So we will loop through all files multiple times.

We should refactor to have a single Walk loop and then conditionally call the SCM / CI specific parsing. 

Filed an issue https://github.com/boostsecurityio/poutine/issues/161